packages/components/src/components/tabs/tabs.tsx

```diff
@@ -52,38 +67,31 @@ export class IfxTabs {
   // needed for smooth border transition
   reRenderBorder() {
     const borderElement = this.el.shadowRoot.querySelector('.active-border') as HTMLElement;
-    if (borderElement && this.tabHeaderRefs[this.activeTabIndex]) {
+    if (borderElement && this.tabHeaderRefs[this.internalActiveTabIndex]) {
       if (this.orientation === 'horizontal') {
 
-        borderElement.style.left = `${this.tabHeaderRefs[this.activeTabIndex].offsetLeft}px`;
-        borderElement.style.width = `${this.tabHeaderRefs[this.activeTabIndex].offsetWidth}px`;
+        borderElement.style.left = `${this.tabHeaderRefs[this.internalActiveTabIndex].offsetLeft}px`;
+        borderElement.style.width = `${this.tabHeaderRefs[this.internalActiveTabIndex].offsetWidth}px`;
         borderElement.style.top = '';
         borderElement.style.height = '';
       } else {
-        borderElement.style.top = `${this.tabHeaderRefs[this.activeTabIndex].offsetTop}px`;
-        borderElement.style.height = `${this.tabHeaderRefs[this.activeTabIndex].offsetHeight}px`;
+        borderElement.style.top = `${this.tabHeaderRefs[this.internalActiveTabIndex].offsetTop}px`;
+        borderElement.style.height = `${this.tabHeaderRefs[this.internalActiveTabIndex].offsetHeight}px`;
         borderElement.style.left = '';
         borderElement.style.width = '';
       }
     }
   }
 
-  @Watch('orientation')
-  onOrientationChange() {
-    this.reRenderBorder()
-  }
 
   // when a slot is removed / added
   @Listen('slotchange')
   onSlotChange() {
     const tabs = this.el.querySelectorAll('ifx-tab');
-    this.tabTitles = Array.from(tabs).map((tab) => tab.getAttribute('header'));
-    this.disabledTabs = Array.from(tabs).map((tab) => tab.getAttribute('disabled'));
-
-    this.tabObjects = this.tabTitles.map((header, index) => {
+    this.tabObjects = Array.from(tabs).map((tab) => {
       return {
-        header: header,
-        disabled: (this.disabledTabs[index] === 'true') // set disabled property based on the corresponding value in the disabledTabs array
+        header: tab?.header === null ? tab?.getAttribute('header') !== null : tab.header,
```

Comment:
Not sure I understand this.. 
If header is null, we then check if null !== null? 🤔